pkg/services/object/acl/eacl/v2/headers.go

```diff
@@ -126,23 +128,32 @@ func (h *cfg) readObjectHeaders(dst *headerSource) error {
 			dst.objectHeaders = objHeaders
 			dst.incompleteObjectHeaders = !completed
 		case
-			*objectV2.GetRangeRequest,
-			*objectV2.GetRangeHashRequest,
-			*objectV2.DeleteRequest:
+			*protoobject.GetRangeRequest,
+			*protoobject.GetRangeHashRequest,
+			*protoobject.DeleteRequest:
 			if h.obj == nil {
 				return errMissingOID
 			}
 
 			dst.objectHeaders = addressHeaders(h.cnr, h.obj)
-		case *objectV2.PutRequest:
-			if v, ok := req.GetBody().GetObjectPart().(*objectV2.PutObjectPartInit); ok {
-				splitHeader := v.GetHeader().GetSplit()
-				if splitHeader == nil || splitHeader.GetSplitID() != nil {
+		case *protoobject.PutRequest:
+			if v, ok := req.GetBody().GetObjectPart().(*protoobject.PutRequest_Body_Init_); ok {
+				if v == nil || v.Init == nil {
+					return errors.New("nil oneof field with heading part")
+				}
+				in := v.Init
+				splitHeader := in.Header.GetSplit()
+				if splitHeader == nil || splitHeader.SplitId != nil {
 					// V1 split scheme or small object, only the received
 					// object's header can be checked
+					mo := &protoobject.Object{
```

Comment:
"mo"?

Reply:
Can be `po`, can be just about anything, short-lived.

Reply:
no problem, i just do not understand what it means